packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart

```diff
@@ -147,6 +166,17 @@ class InspectorTreeController extends Object
     setState(() {
       _root = node;
       _populateSearchableCachedRows();
+
+      ga.select(
+        gac.inspector,
+        'InspectorTreeControllerRootChange',
+        nonInteraction: true,
+        screenMetricsProvider: () => InspectorTreeControllerMetrics(
+          inspectorTreeControllerId: gaId,
+          rootSetCount: ++_rootSetCount,
```

Comment:
why do we need to count how many times the root is set? If we are sending a GA event each time the root is sent, including the count seems redundant

Reply:
this will help us distinguish how many widgets they are seeing throughout the lifecycle of the tab.
The `InspectorTreeControllerInitialized` Item will catch when the controller is first loaded.
Then each subsequent `InspectorTreeControllerRootChange` will determine if they ever get nodes, and if so, if they always continue to get nodes.

For example if we are seeing a lot of `InspectorTreeControllerInitialized` with no follow up root sets, then we know users are seeing blank screens (or else leaving the page before anythign could be loaded)

Another case could be  if rootSetCount>=1 continue to have only 1 row being shown then this could indicate pubRoot directory issues.
We could also see if they are hitting refresh a lot and continually only seeing 1 row.

Continuing to have more events after rootSetCount>=1 would let us get a better peek later in the lifecycle. Perhaps there are blips where they are shown no widgets etc.

Reply:
shouldn't we be able to infer what 'root set' we are on though by the session id and the timestamp of the analytics event? (just a hypothesis, I haven't confirmed this). 

another question: are the inspector tree controllers recreated each time the user switches from the inspector tab to a different DevTools tab and then back?

Reply:
I will have to check this again in the morning but I believe that this resets each time we come back to the tab.

Reply:
I did a double check and the controller continues to exist even when we navigate away. So the count is representative of how many times we have set the root over the lifetime of the controller, not necessarily the lifetime of the tab.


For some added detail, i had a chat with Kenzie to describe what the `rootSetCount` will help with. The count allows us to understand how many rows we are showing to the user at different times of the controller lifecycle. 
We discussed how it would still be possible to get this information without the count, but having the count will make our queries much simpler. So we decided that it is ok to keep.